### PR TITLE
Add ASP.NET SDK documentation

### PR DESCRIPTION
**Pull request by jonashendrickx**
- Added ASP.NET (Identity) SDK documentation
- Reordered the links to list the JavaScript SDK first being more generic.

### CODE REVIEW

File: src/guide/frontend/aspnet.md [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+```
+
+Bootstrapping is now a straightforward process, where the addition of Passwordless to the `IdentityBuilder` becomes a requisite step.
+
+The example below assumes your `appsettings.json` will add all configuration settings under the key `Passwordless`.
+
+```csharp
+builder.Services.AddIdentity<IdentityUser, IdentityRole>()
+    .AddEntityFrameworkStores<PasswordlessContext>()
+    .AddPasswordless(builder.Configuration.GetSection("Passwordless"));
```
Comment (Tyrrrz) [inline src/guide/frontend/aspnet.md:20]: Nit: slightly cleaner

```suggestion
builder.Services
    .AddIdentity<IdentityUser, IdentityRole>()
    .AddEntityFrameworkStores<PasswordlessContext>()
    .AddPasswordless(builder.Configuration.GetSection("Passwordless"));
```

File: src/guide/frontend/aspnet.md [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+  }
+}
+```
+
+Now, we'll need to modify the `WebApplication` object to add the Passwordless routing to make registering, logging in and managing credentials easier.
+
+The endpoints are documented in a later section on this page.
+
+```csharp
+app.MapPasswordless(true);
```
Comment (Tyrrrz) [inline src/guide/frontend/aspnet.md:42]: What does the `true` correspond to here? Maybe add a comment

File: src/guide/frontend/aspnet.md [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+
+Calling `POST /passwordless-register` will create our `IdentityUser` and return a registration token in its response. We will then be able to use that token to create our passkeys.
+
+```html
+@page
+@using Passwordless.Net
+@using Microsoft.Extensions.Options
+@using Passwordless.AspNetCore
+@model RegisterModel
+@inject IOptions<PasswordlessAspNetCoreOptions> PasswordlessOptions;
```
Comment (Tyrrrz) [inline src/guide/frontend/aspnet.md:57]: Nit: slightly cleaner

```suggestion
@page

@using Passwordless.Net
@using Microsoft.Extensions.Options
@using Passwordless.AspNetCore

@model RegisterModel

@inject IOptions<PasswordlessAspNetCoreOptions> PasswordlessOptions;
```

File: src/guide/frontend/aspnet.md [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+
+@{
+    ViewData["Title"] = "Register";
+}
+<h1>@ViewData["Title"]</h1>
+
+@{
+    var canAddPasskeys = ViewData["CanAddPasskeys"] != null && (bool)ViewData["CanAddPasskeys"];
```
Comment (Tyrrrz) [inline src/guide/frontend/aspnet.md:65]: Can be simplified

```suggestion
    var canAddPasskeys = ViewData["CanAddPasskeys"] is true;
```

File: src/guide/frontend/aspnet.md [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+    }
+
+    public async Task OnPostAsync(FormModel form, CancellationToken cancellationToken)
+    {
+        if (!ModelState.IsValid) return;
+        _logger.LogInformation("Registering user {username}", form.Username);
+        ViewData["CanAddPasskeys"] = true;
+    }
+    
+    public FormModel Form { get; init; } = new();
```
Comment (Tyrrrz) [inline src/guide/frontend/aspnet.md:146]: Fix identation

```suggestion
public IActionResult OnGet()
{
    if (HttpContext.User.Identity is { IsAuthenticated: true })
    {
        return LocalRedirect("/");
    }

    return Page();
}

public async Task OnPostAsync(FormModel form, CancellationToken cancellationToken)
{
    if (!ModelState.IsValid) return;
    _logger.LogInformation("Registering user {username}", form.Username);
    ViewData["CanAddPasskeys"] = true;
}

public FormModel Form { get; init; } = new();
```

File: src/guide/frontend/aspnet.md [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+            body: JSON.stringify(loginRequest)
+        });
+        
+        if (loginResponse.ok) {
+            console.log('login successful: ' + await loginResponse.text());
+            
+            // redirect to authorized page /Authorized/HelloWorld
+            window.location.href = '/Authorized/HelloWorld';        }
+    }
+    login();
```
Comment (Tyrrrz) [inline src/guide/frontend/aspnet.md:221]: Fix identation

```suggestion
        async function login() {
            const alias = document.getElementById("email").value;
            const p = new Passwordless.Client({
                apiKey: "@PasswordlessOptions.Value.ApiKey",
                apiUrl: "@PasswordlessOptions.Value.ApiUrl"
            });
            const loginPasskeyResponse = await p.signinWithAlias(alias);
            if (!loginPasskeyResponse) {
                return;
            }
            const loginRequest = {
                token: loginPasskeyResponse.token
            };
            const loginResponse = await fetch('/passwordless-login', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json'
                },
                body: JSON.stringify(loginRequest)
            });
        
            if (loginResponse.ok) {
                console.log('login successful: ' + await loginResponse.text());
        
                // Redirect to authorized page /Authorized/HelloWorld
                window.location.href = '/Authorized/HelloWorld';
            }
        }
        
        login();
```

File: src/guide/frontend/aspnet.md [hunk 7]
```diff
@@ -0,0 +1,294 @@
...
+    }
+
+    public async Task OnPostAsync(LoginForm form, CancellationToken cancellationToken)
+    {
+        if (!ModelState.IsValid) return;
+        _logger.LogInformation("Logging in user {email}", form.Email);
+        ViewData["CanLogin"] = true;
+    }
+
+    public LoginForm Form { get; } = new();
```
Comment (Tyrrrz) [inline src/guide/frontend/aspnet.md:245]: Fix identation

```suggestion
public IActionResult OnGet()
{
    if (HttpContext.User.Identity is { IsAuthenticated: true })
    {
        return LocalRedirect("/");
    }
    return Page();
}

public async Task OnPostAsync(LoginForm form, CancellationToken cancellationToken)
{
    if (!ModelState.IsValid) return;
    _logger.LogInformation("Logging in user {email}", form.Email);
    ViewData["CanLogin"] = true;
}

public LoginForm Form { get; } = new();
```

File: src/guide/frontend/aspnet.md [hunk 8]
```diff
@@ -0,0 +1,294 @@
...
+| Register__Discoverable       | true                        | [false/true] Discoverable Credentials store private keys and associated data in the authenticator's memory instead of on a server. This eliminates the need for the server to send the credential to the authenticator for decryption, reducing the reliance on usernames and passwords for identity verification. |
+| Register__UserVerification   | Preferred                   | [Discouraged/Preferred/Required] A WebAuthn Relying Party may require user verification for some of its operations but not for others, and may use this type to express its needs.                                                                                                                                 |
+                                                                                                                                                                                                                                                                                       |
+# Advanced
```
Comment (Tyrrrz) [inline src/guide/frontend/aspnet.md:289]:
```suggestion

# Advanced
```

File: src/guide/frontend/aspnet.md (hunk 2 above)
Comment (abergs) [reply src/guide/frontend/aspnet.md:42]: Or we can name the argument? `(enableRegisterEndpoint: true)`

**Review by abergs — APPROVED** [review]

**Comment by abergs** [comment]
Nice work!

File: src/.vuepress/config.js [hunk 9]
```diff
@@ -80,7 +80,9 @@ module.exports = {
               path: '/guide/frontend',
               collapsable: true,
               children: [
+
                 ['/guide/frontend/javascript', 'JavaScript Client'],
+                ['/guide/frontend/aspnet', 'ASP.NET'],
```
[the remark this answers is not in the document]
Comment (jonashendrickx) [reply src/.vuepress/config.js:85]: @Tyrrrz Maybe we can create a separate 'Frameworks' area? I suppose we may need one for Spring Boot too later on.

[the remark this answers is not in the document]
Comment (abergs) [reply src/.vuepress/config.js:84]: @jonashendrickx @Tyrrrz Agree'd that we'd having ASP.NET under Frontend is not ideal. Let's restructure the docs to make more sense now that we have multiple frameworks and languages, but not in this PR.

File: src/guide/frontend/aspnet.md [hunk 10]
```diff
@@ -0,0 +1,303 @@
...
+  }
+}
+```
+
+Now, we'll need to modify the `WebApplication` object to add the Passwordless routing to make registering, logging in and managing credentials easier.
+
+The endpoints are documented in a later section on this page.
+
+```csharp
+app.MapPasswordless(true);
```
Comment (abergs) [inline src/guide/frontend/aspnet.md:43]:
```suggestion
app.MapPasswordless(enableRegisterEndpoint: true);
```

**Pull request merged**